Platon/backend/app/profile_management/views.py

```diff
@@ -107,6 +111,174 @@ def delete(user_id,self):
+
+        @api.doc(responses={201: 'Successfully Created', 400: 'Wrong Input Format', 401: 'Authentication Problem',
+                            500: 'Database Connection Problem'})
+        @api.expect(jobs_post_parser)
+        def post(self):
+            """
+                Creates a new job
+            """
+            form = JobsPostForm(request.form)
+            if form.validate():
+                new_job = Jobs(form.name.data)
+                try:
+                    db.session.add(new_job)
+                    db.session.commit()
+                except:
+                    return make_response(jsonify({'error': 'Database Connection Problem'}), 500)
+                return make_response(jsonify({'msg': ' Successfully added'}), 201)
+            else:
+                return make_response(jsonify({'error': 'Wrong input format'}), 400)
+
+        @api.doc(responses={201: 'Successfully Updated', 400: 'Wrong Input Format', 401: 'Authentication Problem',
+                            500: 'Database Connection Problem'})
+        @api.expect(jobs_put_parser)
+        def put(self):
+            """
+                Updates a job
+            """
+            form = JobsPutForm(request.form)
+            if form.validate():
+                try:
+                    job = Jobs.query.filter(Jobs.id == form.id.data).first()
+                    if job is None:
+                        return make_response(jsonify({'error': 'Please give an appropriate job ID'}), 400)
+                    job.name = form.name.data if form.name.data != '' else job.name
+                    db.session.commit()
+                except:
+                    return make_response(jsonify({'error': 'Database Connection Problem'}), 500)
+                return make_response(jsonify({'msg': ' Successfully changed'}), 201)
+            else:
+                return make_response(jsonify({'error': 'Wrong input format'}), 400)
+
+        @api.doc(responses={200: 'Successfully Deleted', 400: 'Wrong Input Format', 401: 'Authentication Problem',
+                            500: 'Database Connection Problem'})
+        @api.expect(jobs_delete_parser)
+        def delete(self):
+            """
+                Deletes a job
+            """
+            form = JobsDeleteForm(request.form)
+            if form.validate():
+                try:
+                    job = Jobs.query.filter(Jobs.id == form.id.data).first()
+                    if job is None:
+                        return make_response(jsonify({'error': 'Job does not exist'}), 400)
+                    db.session.delete(job)
+                    db.session.commit()
+                    return make_response(jsonify({'msg': 'Successfully Deleted'}), 200)
+                except:
+                    return make_response(jsonify({'error': 'Database Connection Problem'}), 500)
+            else:
+                return make_response(jsonify({'error': 'Wrong input format'}), 400)
```

Comment:
What is the purpose of these 2 endpoints? I really do not understand the aim, I think frontend or android will not use these endpoints.

Reply:
I added them in case of a user enters wrong input and she wants to change or delete them so that database holds no unnecessary data.